design/FY2021/ReportIndividualInfiltrationObject.md

```diff
@@ -0,0 +1,96 @@
+﻿# Add ZoneInfiltration Output Variables for Individual objects(Issue #8209)
+================
+
+**Yueyue Zhou, NREL**
+
+ - 25 June 2021
+  
+
+## Background ##
+
+Excerpts lifted from [issue #8209](https://github.com/NREL/EnergyPlus/issues/8209)
+
+ - There can be one infiltration object serving multiple zones, and there can be multiple infiltration objects serving the same zone. Currently the output variables only report at zone level, which is good for the first case. However, for the second case, there's no way to obtain, e.g., Sensible Heat Gain Energy specific to each object.
+ - Some related discussion in [Add Space PR](https://github.com/NREL/EnergyPlus/pull/8394#discussion_r530585696):
+	 - Dan Macumber: Is it possible to make infiltration or ventilation at the space level? That would simplify input for cases where ventilation requirements are a function of space type.
+	 - Jason Degraw: I think it is possible, I think we probably just want to avoid having both (e.g. SpaceInfiltration _and_ ZoneInfiltration). Moving everything to the space level will probably be more work, but if that's the right solution then we should do that.
+	 - Scott Horowitz: Whether it's SpaceInfiltration or ZoneInfiltration, it'd be nice to be able to get individual outputs for each object. With current EnergyPlus, you can have multiple ZoneInfiltration objects in a zone, but can only get aggregate outputs (i.e., the key for Zone Infiltration Sensible Heat Gain Energy is a zone object, not an infiltration object).
+	 - Richard Raustad: Should probably have outputs for both space and zone to show individual as well as aggregated.
+	 - Edwin Lee: I agree with this: limit to one input approach but output at both the space and zone level.
+ 
+ 
+## E-mail and  Conference Call Conclusions ##
+
+*None yet.*
+
+## Overview ##
+
+This work will implement code changes that calculate infiltration output variables at individual levels. Proposed list of variables to report is: 
+
+ - Infiltration Sensible Heat Loss Energy
+ - Infiltration Sensible Heat Gain Energy
+ - Infiltration Latent Heat Loss Energy
+ - Infiltration Latent Heat Gain Energy
+ - Infiltration Total Heat Loss Energy
+ - Infiltration Total Heat Gain Energy
+ - Infiltration Current Density Volume Flow Rate
+ - Infiltration Standard Density Volume Flow Rate
+ - Infiltration Current Density Volume
+ - Infiltration Standard Density Volume
+ - Infiltration Mass
+ - Infiltration Mass Flow Rate
+ - Infiltration Air Change Rate
+ 
+Considering the mapping of multiple infiltration and multiple zone, it might require two-level reporting, eg:
+1. **Infiltration1** is serving ThermalZoneList where **ThermalZone1** and **ThermalZone2** are included
+2. **Infiltration2** is serving **ThermalZone1** only
```

Comment:
Existing reports use "Zone Infiltration something". 

Why not add:
"Space Infiltration something"
"Total Infiltration something" - object level reporting

Reply:
Thanks, @rraustad .
For space level reporting, is it scoped to be added in [AddSpace PR](https://github.com/NREL/EnergyPlus/pull/8394)?
For the object-level reporting, just to clarify, it is not actually grouping all the zones that one infiltration serves, instead it will be reported as `ZoneName InfiltrationName: Infiltiration something`. This is based on how `People` is reported. So for those, it's not supposed to add `Total` to the variable names. Do you suggest other than individual variable reporting, we can group them by infiltration and add `Total Infiltration something` ?

Reply:
Re: space outputs, yes, those will be added as part of #8394 .

Re: "mapping of multiple infiltration and multiple zone" I'm not sure what this is referring to. Is this talking about the use of a ZoneList Name? If so, the input processing already names each instance automatically as "Zone Name Object Name", so you don't need to do anything special here. Just use the `Infiltration(loop).Name` in the call to `SetupOutput` and it will be fine.
This happens in [`CheckCreatedZoneItemName](https://github.com/NREL/EnergyPlus/blob/42846a95e03cabcc18ef22f9778ca58ae6468128/src/EnergyPlus/General.cc#L1763) when it is called from [here](https://github.com/NREL/EnergyPlus/blob/42846a95e03cabcc18ef22f9778ca58ae6468128/src/EnergyPlus/HeatBalanceAirManager.cc#L829-L841) and similar places during input processing.

Reply:
I guess you can subset infiltration like that, by object and zone, I'm just not sure how useful that will be. I did not realize, except from your NFP, that the people object did the same thing. I would think users would want to know how many people are in a zone instead of how many people are in this zone based on this specific people object. For infiltration (or people) I guess it might be useful to know how much infiltration was in a zone due to the Design Flow Rate Calculation Method specified in each infil object. This is one of those things where I would like to know what user needs are and how often they use these subset reporting methods.

Reply:
Thanks, @mjwitte . Yes it means the zonelist name. One infiltration can serve multiple zones by specifying zonelist name, and one zone can have multiple infiltrations (they're grouped to report in existing approach.) Sorry for the confusion. And thanks for pointing to the `CheckCreatedZoneItemName`. I realized it when I was investigating `People` reporting which is essentially what we want to achieve. I will mostly follow how `People` was implemented to keep them consistent.